qcodes/tests/test_autoloadable_channels.py

```diff
@@ -0,0 +1,186 @@
+            new_channel += 1
+
+        kwargs = {
+            "name": f"channel{new_channel}",
+            "channel": new_channel
+        }
+
+        return kwargs
+
+    def __init__(
+            self,
+            parent: Union[Instrument, 'InstrumentChannel'],
+            name: str,
+            channel: int,
+            existence: bool = False,
+            channel_list: 'AutoLoadableChannelList' = None,
+    ) -> None:
+
+        super().__init__(parent, name, existence, channel_list)
+        self._channel = channel
+
+        self.add_parameter(
+            "hello",
+            get_cmd=f":INST:CHN{self._channel}:HLO"
+        )
+
+    def _create(self)->None:
+        """Create the channel on the instrument"""
+        self.parent.write(f":INST:CHN:ADD {self._channel}")
+
+    def _delete(self)->None:
+        """Remove the channel from the instrument"""
+        self.write(f":INST:CHN:DEL {self._channel}")
+
+
+class TestInstrument(Instrument):
+    """
+    This dummy instrument allows the creation and deletion of
+    channels
+    """
+
+    def __init__(self, name):
+        super().__init__(name)
+
+        self._backend = TestInstrumentBackend()
+
+        self.add_parameter(
+            "channel_catalog",
+            get_cmd=":INST:CHN:CAT",
+        )
+
+    def write_raw(self, cmd: str):
+        return self._backend.send(cmd)
+
+    def ask_raw(self, cmd: str):
+        return self._backend.send(cmd)
+
+
+class DummyInstrumentWithChannelList(TestInstrument):
+    def __init__(self, name):
+        super().__init__(name)
+
+        channels = AutoLoadableChannelList(self, "channels", SimpleTestChannel)
+        self.add_submodule("channels", channels)
+
+
+def test_sanity():
+
+    instrument = TestInstrument("instrument")
+    channels = SimpleTestChannel.load_from_instrument(instrument)
+    assert len(channels) == 3
+    assert channels[0].hello() == "Hello from channel 1"
+    assert channels[1].hello() == "Hello from channel 2"
+    assert channels[2].hello() == "Hello from channel 4"
+
+    new_channel_kwargs = SimpleTestChannel.get_new_instance_kwargs(instrument)
+    new_channel = SimpleTestChannel(instrument, **new_channel_kwargs)
+
+    with pytest.raises(RuntimeError):
+        new_channel.hello()  # We have only instantiated a channel class, we
+        # have yet to create it on the instrument
+
+    new_channel.create()
+    assert new_channel.hello() == "Hello from channel 3"
+    new_channel.delete()
+
+    with pytest.raises(RuntimeError):
+        new_channel.hello()  # We have deleted the channel and it should no longer be
+        # available
+
+
+def test_channels_list():
+    instrument = DummyInstrumentWithChannelList("instrument2")
+    assert instrument.channels[0].hello() == "Hello from channel 1"
+    assert instrument.channels[1].hello() == "Hello from channel 2"
+    assert instrument.channels[2].hello() == "Hello from channel 4"
+
+    new_channel = instrument.channels.add()
```

Comment:
what happens when the instrument is closed but there are still instances of channels lying around? their methods should be protected against this in order to help in debugging possible weird scenarios.

Reply:
That is indeed a good point, but the same can be said for any submodule, even existing channel classes. What if for the SR860 I do something like: 

```python 
lockin = SR860("lockin", "GPIB0::4::INSTR")
data_channel = lockin.data_channel_3
lockin.close()
```
You will have exactly the same problem. So this is a problem that should be fixed in another PR.

Reply:
ok, i'll make an issue